### ISSUE TITLE
NVIDIA open-sourced its GPU kernel module. Time to integrate it into xanmod?

### ISSUE THREAD

**Issue by enihcam** [xanmod/linux#264]
https://github.com/NVIDIA/open-gpu-kernel-modules

**Comment by ThatOneCalculator** [comment]
It's by no means ready, let alone usable. It's incomplete, contains MANY binary blobs, and performs terribly, and crashes a bunch.

**Comment by enihcam** [comment]
> means ready, let alone usable. It's incomplete, co

Define 'usable', at least it is ready for Archlinux official.

https://archlinux.org/packages/extra/x86_64/nvidia-open/

**Comment by enihcam** [comment]
> It's incomplete, contains MANY binary blobs, and performs terribly, and crashes a bunch.

Where is the 'MANY binary blobs':
https://archlinux.org/packages/extra/x86_64/nvidia-open/files/

**Comment by ThatOneCalculator** [comment]
CUDA, Vulkan, OpenGL, Tensorcore drivers, ML acceleration, and more are all closed source.

**Comment by enihcam** [comment]
> CUDA, Vulkan, Tensorcore drivers, ML acceleration, and more are all closed source in the "open" driver.

linux kernel is not for those, right?

**Comment by ThatOneCalculator** [comment]
The Linux kernel isn't for incorporating a driver where most of the important code is closed source :laughing:

**Comment by enihcam** [comment]
> The Linux kernel isn't for incorporating a driver where most of the important code is closed source 😆

i agree but why this relates to integrating the open-part into the kernel?

`The open-source kernel-mode driver works with the same firmware and the same user-mode stacks such as CUDA, OpenGL, and Vulkan.` https://developer.nvidia.com/blog/nvidia-releases-open-source-gpu-kernel-modules/

**Comment by ThatOneCalculator** [comment]
Because the "open part" is nothing more than a surface-level userland gateway to closed source software.

**Comment by enihcam** [comment]
> Because the "open part" is nothing more than a surface-level userland gateway to closed source software.

well, in that way xanmod kernel has lots of 'nothing more' already. e.g. winesync, Android ashmem and binder, etc. they are for running the 'closed source software'.
